Sloth.Core/Extensions/AggieEnterpriseExtensions.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace Sloth.Core.Extensions
+{
+    public static class AggieEnterpriseExtensions
+    {
+        public static string StripToErpName(this string value, int maxLength = Int32.MaxValue)
+        {
+            // Erp names can only contain letters, numbers, hyphen, spaces and underscores.
+            const string regexStr = @"[^0-9a-zA-Z\ \_\-\']+";
+
+            return value.SafeTruncate(maxLength).SafeRegexRemove(regexStr);
```

Comment:
@srkirkland Do the regex remove before the truncate. It may shorten it.

Reply:
good idea